### PR TITLE
Add support for restart field

### LINKED ISSUE #1056: Support for "jobs" on Okteto Stacks
**Is your feature request related to a problem? Please describe.**
When I deploy my stack, some of my services need data initialization. This is usually implemented in Kubernetes using jobs.

**Describe the solution you'd like**
A friendly way to define Jobs using a `jobs` section.

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes #1056 

## Proposed changes
-  Add support for restart field. 
-  With this change you can create jobs in docker-compose

### CODE REVIEW

File: pkg/cmd/stack/translate.go [hunk 1]
```diff
@@ -388,6 +389,54 @@ func translateStatefulSet(svcName string, s *model.Stack) *appsv1.StatefulSet {
...
+
+	return &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        svcName,
+			Namespace:   s.Namespace,
+			Labels:      translateLabels(svcName, s),
+			Annotations: translateAnnotations(svc),
+		},
+		Spec: batchv1.JobSpec{
+			Parallelism: pointer.Int32Ptr(svc.Replicas),
```
Comment (pchico83) [inline pkg/cmd/stack/translate.go:405]: I think `Replicas` should translate to `Completions`

File: pkg/cmd/stack/translate.go [hunk 2]
```diff
@@ -388,6 +389,54 @@ func translateStatefulSet(svcName string, s *model.Stack) *appsv1.StatefulSet {
...
+			Selector: &metav1.LabelSelector{
+				MatchLabels: translateLabelSelector(svcName, s),
+			},
+			ManualSelector: pointer.BoolPtr(true),
```
Comment (pchico83) [inline pkg/cmd/stack/translate.go:409]: should we leave this empty?

File: pkg/model/stack_serializer.go [hunk 3]
```diff
@@ -138,20 +138,27 @@ type ServiceRaw struct {
...
-	RestartPolicy  *WarningType `yaml:"restart_policy,omitempty"`
 	RollbackConfig *WarningType `yaml:"rollback_config,omitempty"`
 	UpdateConfig   *WarningType `yaml:"update_config,omitempty"`
 }
 
+type RestartPolicyRaw struct {
+	Condition string `yaml:"condition,omitempty"`
+
+	Delay       *WarningType `yaml:"delay,omitempty"`
+	MaxAttempts *WarningType `yaml:"max_attempts,omitempty"`
```
Comment (pchico83) [inline pkg/model/stack_serializer.go:159]: This could translate to `BackoffLimit`

File: pkg/cmd/stack/translate.go (hunk 1 above)
Comment (pchico83) [reply pkg/cmd/stack/translate.go:405]: `Parallelism` should be always 1

File: pkg/cmd/stack/translate.go [hunk 4]
```diff
@@ -388,6 +389,54 @@ func translateStatefulSet(svcName string, s *model.Stack) *appsv1.StatefulSet {
...
+			Template: apiv1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels:      translateLabels(svcName, s),
+					Annotations: translateAnnotations(svc),
+				},
+				Spec: apiv1.PodSpec{
+					RestartPolicy:                 svc.RestartPolicy,
```
Comment (pchico83) [inline pkg/cmd/stack/translate.go:416]: This should be `Never`.  The job will create another pod if this one fails

**Comment by pchico83** [comment]
@jLopezbarb could we create an e2e test for this?

File: pkg/cmd/stack/translate.go (hunk 2 above)
Comment (jLopezbarb) [reply pkg/cmd/stack/translate.go:409]: If you leave `ManualSelector` empty you can not add any selector, and when listing the jobs we can not see if it's from another stack

Comment (pchico83) [reply pkg/cmd/stack/translate.go:409]: From the docs:
```
	// manualSelector controls generation of pod labels and pod selectors.
	// Leave `manualSelector` unset unless you are certain what you are doing.
	// When false or unset, the system pick labels unique to this job
	// and appends those labels to the pod template.  When true,
	// the user is responsible for picking unique labels and specifying
	// the selector.  Failure to pick a unique label may cause this
	// and other jobs to not function correctly.  However, You may see
	// `manualSelector=true` in jobs that were created with the old `extensions/v1beta1`
	// API.
	// More info: https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#specifying-your-own-pod-selector
	// +optional
	ManualSelector *bool `json:"manualSelector,omitempty" protobuf:"varint,5,opt,name=manualSelector"`
```
We don't need control on the pod labels, no? we only need control on the job labels to do queries, right?

**Comment by jLopezbarb** [comment]
I updated the e2e test for compose adding an initialization for the queue

**Comment by pchico83** [comment]
> I updated the e2e test for compose adding an initialization for the queue

is the e2e test ready to test the job is created/and deleted? would the test fail if the queue is not properly initialized?

**Review by pchico83 — APPROVED** [review]
I think we should update the compose test to check that jobs are deleted:
https://github.com/okteto/okteto/blob/master/integration/stacks_test.go#L237

File: pkg/cmd/stack/deploy.go [hunk 5]
```diff
@@ -296,7 +328,7 @@ func waitForPodsToBeRunning(ctx context.Context, s *model.Stack, c *kubernetes.C
 			return err
 		}
 		for i := range podList {
-			if podList[i].Status.Phase == apiv1.PodRunning {
+			if podList[i].Status.Phase == apiv1.PodRunning || podList[i].Status.Phase == apiv1.PodSucceeded {
```
Comment (pchico83) [inline pkg/cmd/stack/deploy.go:331]: If a pod is in `PodFailed` we should fail and return an error.

**Review by pchico83 — APPROVED** [review]

**Pull request merged**